DriverLoader.cpp

```diff
@@ -91,12 +93,19 @@ namespace vive {
             throw CouldNotLoadEntryPoint();
         }
         factory_ = reinterpret_cast<DriverFactory>(proc);
-#elif defined(OSVR_MACOSX)
-#error                                                                         \
-    "Implementation incomplete! Load dynamic library here and retrieve entry point function here!"
-#elif defined(OSVR_LINUX)
-#error                                                                         \
-    "Implementation incomplete! Load dynamic library here and retrieve entry point function here!"
+#elif defined(OSVR_LINUX) || defined(OSVR_MACOSX)
+        impl_->driver_ = dlopen(driverFile.c_str());
+        if (!impl_->driver_) {
+            reset();
+            throw CouldNotLoadDriverModule(dlerror());
+        }
+
+        auto proc = dlsym(impl_->driver_, ENTRY_POINT_FUNCTION_NAME);
```

Comment:
Do we need to call `dlerror()` first here unconditionally? I'm looking at some of the code that loads OSVR plugins and it looks like that's what I do https://github.com/OSVR/libfunctionality/blob/master/src/libfunctionality/LoadPluginLibdl.h#L62